### PR TITLE
updating pre-commit and adding pyproject.toml

### PR DESCRIPTION
**Pull request by GuillermoFidalgo**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -20,18 +20,8 @@ repos:
...
-  - repo: "https://github.com/pre-commit/pygrep-hooks"
-    rev: v1.9.0
-    hooks:
-      - id: python-check-blanket-noqa
-      - id: python-check-blanket-type-ignore
-      - id: python-use-type-annotations
-  - repo: "https://github.com/pycqa/isort"
-    rev: 5.10.1
-    hooks:
-      - id: isort
```
Comment (mambelli) [inline .pre-commit-config.yaml:32]: Are these checks already included in ruff?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -49,14 +39,8 @@ repos:
...
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: "v0.4.5"
     hooks:
-      - id: pylint
-  - repo: "https://github.com/fsfe/reuse-tool"
-    rev: v0.14.0
-    hooks:
-      - id: reuse
-        additional_dependencies:
-          - python-debian==0.1.40
```
Comment (mambelli) [inline .pre-commit-config.yaml:62]: Why did you remove the license checking?

**Review by mambelli — CHANGES_REQUESTED** [review]

**Review by mambelli — APPROVED** [review]

**Comment by mambelli** [comment]
Approving to get the repo as you like for the class. I may have to check and revert some changes in the precommit later

**Comment by GuillermoFidalgo** [comment]
Sorry I didn't have a chance to answer the message. Today as we expected was very hectic.

I simply deleted some pre-commits and checked what worked. I remember reading a problem in the logs with isort. So I removed that because it is in ruff. 
But I remember that the licence check see was still working. At least it required the license on the toml file I added so I don't know what it does.


It could just as well be reimplemented.
